### PR TITLE
Final prep for v0.8.1 release

### PR DESCRIPTION
**Pull request by ascillitoe**
TODO:

- [x] Deprecate Python 3.6.
- [x] Add bounds for `numba` version.
- [x] https://github.com/SeldonIO/alibi-detect/issues/428
- [x] CHANGELOG.md

### CODE REVIEW

**Comment by ascillitoe** [comment]
@mauicv taking this off draft to run tests, it's not actually ready yet.

**Comment by ascillitoe** [comment]
Wrt to consolidating the examples dependencies, a few considerations:

- I've only included dependencies from notebooks that are included in tests. For example, `wilds` isn't included, since the `cd_online_camelyon.ipynb` and `cd_text_amazon.ipynb` notebooks are excluded from the tests.
- I've got rid of the `"examples"` in `extras_require`, and moved it to `testing/requirements.txt`. Thinking behind this is: 1) we don't actually expose the `pip install .[examples]` capability anywhere in the `alibi-detect` docs, and currently all necessary `!pip install` calls are documented in the notebooks anyway. 2) Adding dependencies to `setup.py` makes them more formal dependencies in my view (even if they're in `extras_require`), and brings them under dependabot etc. I'm not sure this is desirable for packages like `TimeSynth` which aren't in PyPi. 
- I've gone with `testing/requirements.txt` instead of `requirements/examples.txt`, since this list of dependencies is private/internal i.e. its only intended for testing usage, and doesn't include all dependencies used in examples (such as `wilds`).

@jklaise @mauicv The above are all relatively opinionated decisions, so happy to change if you disagree!

**Review by mauicv — APPROVED** [review]
All looks good to me!

File: testing/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Dependencies installed (with !pip install) in example notebooks are listed below.
+# Note: These dependencies are installed during the `test_all_notebooks` and 
+# `test_changed_notebooks` github workflows. The dependencies of notebooks excluded 
+# from testing (see `EXCLUDE_NOTEBOOKS` in `testing/test_notebooks.py`) are not 
+# included here.
+alibi==0.6.2
```
Comment (jklaise) [inline testing/requirements.txt:6]: Any reason to pin this?

Comment (jklaise) [reply testing/requirements.txt:6]: Yea I think latest stable (via `alibi<1.0.0`) is good, also might catch some bugs between libraries.

Comment (ascillitoe) [reply testing/requirements.txt:6]: Ok sounds good!

**Review by jklaise — APPROVED** [review]

**Pull request merged**